### PR TITLE
feat: Add functionality for inviting reviewers to submissions

### LINKED ISSUE #1256: Reviewer Invitation Functionality
### Problem/Why
While user interface updates are being implemented on the Submission Details page in issue #1399 to support the invitation of reviewers and review coordinators, these features are non-functional.

### Goal/What
Implement functionality that allows a review coordinator, editor, or administrator, to send an email inviting a user to be assigned as a reviewer or review coordinator of a submission.

### Solution/How
Create the email, the functionality to send the email, and functionality to accept the invitation. If the user does not already exist in the system, the user should be staged immediately. A unique, default username and password should be generated for this staged user. Add an attribute to the user model `staged` to track whether or not a user is staged. 

This component should only be interactive for review coordinators, editors, publication administrators, and application administrators. The email should include a descriptive message about the invitation and a link to accept the invitation. 

### To Do / Acceptance Criteria

- [x] Create template for invitation email
- [x] Add functionality to stage a created user
- [x] Add functionality to send an invitational email to a supplied email address
- [x] Add functionality that unstages a user once an invitation is accepted

### Review Checklist

- [ ] A user that does not exist in the application can be invited as a reviewer to a submission

### Accessibility Checklist

- [ ] The language in the email message uses [plain language](https://sheribyrnehaber.com/plain-language-essential-for-content-accessibility/)
- [ ] Usage of color passes at least AA WCAG contrast ratio
- [ ] Font sizes are at least 16px

### PR DESCRIPTION
**Pull request by gmeben**
This pull request

* Enables a user with sufficient privilege to invite reviewers and review coordinators to a submission from the Submission Details page
* Adds the `staged` property to `Users`
* Adds mutations to the GraphQL endpoint
  * `acceptSubmissionInvite`
  * `inviteReviewer`
  * `inviteReviewCoordinator`
* Resolves issues with the [previous splitting](https://github.com/MESH-Research/CCR/pull/1487/files) of `AssignedUsersComponent` into `AssignedSubmissionUsers` and `AssignedPublicationUsers`

Closes #1256

### CODE REVIEW

**Review by wreality — APPROVED** [review]
Just make we're checking that a user has the correct permission when calling the graphql invite endpoints.  Any other notes, I'm happy to kibitz about as needed.

File: backend/app/Models/SubmissionInvitation.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    /**
+     * Set a default token and expiration upon creation
+     *
+     * @return void
+     */
+    protected static function booted()
+    {
+        static::created(function (SubmissionInvitation $invite) {
+            $invite->token = Str::uuid()->toString();
+            $invite->expiration = Carbon::now()->addDays(5)->toDateTimeString();
```
[the remark this answers is not in the document]
Comment (gmeben) [reply backend/app/Models/SubmissionInvitation.php:32]: I considered this in my initial approach, but you've convinced me that the versatility it offers is worth implementing. When expiration is made functional, I'm going to ensure that this is refactored the way you suggested.

File: backend/app/GraphQL/Mutations/InviteSubmissionUser.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+     * @return \App\Models\Submission
+     */
+    public function acceptInvite($_, $args)
+    {
+        $invite = SubmissionInvitation::where('token', $args['token'])->firstOrFail();
+        $invite->accepted_at = Carbon::now()->toDateTimeString();
+        $invite->save();
+        $user = User::where('email', $invite->email)->firstOrFail();
+        $user->staged = null;
+        $user->save();
```
[the remark this answers is not in the document]
Comment (gmeben) [reply backend/app/GraphQL/Mutations/InviteSubmissionUser.php]: This has been refactored as suggested.

File: backend/app/GraphQL/Mutations/InviteSubmissionUser.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                'id' => $submission->id,
+            ],
+            'inviter' => [
+                'name' => $auth_user->name,
+                'username' => $auth_user->username,
+            ],
+            'message' => $args['message'],
+            'token' => $invite->token,
+        ];
+        Notification::send($coordinator, new InviteReviewCoordinator($notification_data));
```
[the remark this answers is not in the document]
Comment (gmeben) [reply backend/app/GraphQL/Mutations/InviteSubmissionUser.php]: This has been refactored as suggested.

**Pull request merged**